src/btree.rs

```diff
@@ -3,15 +3,13 @@ use crate::oram::BUCKET_SIZE;
 #[derive(Debug, Clone, Default)]
 pub struct BTree {
     pub(super) root: Option<Box<Node>>,
-    nb_blocks: usize,
     height: u16,
 }
 
 impl BTree {
     pub fn init_new(dummies: &mut Vec<DataItem>, nb_blocks: usize) -> BTree {
```

Comment:
The idiomatic way of passing reference on a sequence is using a slice (in order to abstract the backing structure and allow more flexibility):
```suggestion
    pub fn init_new(dummies: &mut [DataItem], nb_blocks: usize) -> BTree {
```

Reply:
But there would be no methode to remove element as in line 55 ?
```                    node.set_bucket_element(dummies.remove(j), i);
```

Reply:
True. I didn't see that. This comment still holds for the other functions. For this one, you may want to use another structure instead (linked list?) to avoid the `O(n)` cost in the worst case to call `.remove()`.

To avoid going trough your sequence of items several times, you could go through once and store the indices of (or pointers to) the elements with a matching path.

Reply:
Maybe switching to a linked list would be better ? In the meantime I think of using `swap_remove()` method in $O(1)$

Reply:
Great, I didn't know this method. I was about to propose you to set the path to `u16::MAX` or something like that to avoid considering it in a later time, but `swap_remove()` is much better since it reduces the list size.